### PR TITLE
Add script to plot experiment RMSE, bias, counts from GSI stat file

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
@aerorahul had a two-step workflow to parse and plot GSI stat files, and later introduced a GSI stat Python class to simplify this process. This PR adds a script that uses the PyGSI GSI stat class to read gsistat files, and then plots bias, RMSE, and counts for a control and N experiments (currently just O-F profiles of radiosondes, but should be easily adaptable in the future)

NOTE: this will likely fail if you run 'out of the box', please make sure you add the path of this cloned repo (/path/to/clone/PyGSI) to your $PYTHONPATH, so when you run the script, the python interpreter can find the files needed by 'import pygsi'

This closes issue #30

### CODE REVIEW

File: scripts/plot_gsi_stat_exp.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    for exp, gsistats in zip(labels, args.gsistats):
+        rmses[exp] = {}; counts[exp] = {}; biases[exp] = {}
+        for gsistat, cycle in zip(statfiles, cycles):
+            rmses[exp][cycle] = {}; counts[exp][cycle] = {}; biases[exp][cycle] = {}
+            inputfile = os.path.join(gsistats, gsistat)
+            try:
+                gdas = GSIstat(inputfile, cycle)
+            except FileNotFoundError:
+                print('Error: One or more expected gsistat files missing, check dates and paths!')
+                sys.exit(1)
```
Comment (aerorahul) [inline scripts/plot_gsi_stat_exp.py:131]: use:
```
raise FileNotFoundError(f'Unable to find {inputfile} for cycle {cycle}')
```

File: scripts/plot_gsi_stat_exp.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        mydate = mydate + datetime.timedelta(hours=6)
+    return statfiles, cycles
+
+if __name__ == '__main__':
+    # get command line arguments
+    parser = argparse.ArgumentParser(description=('Plots a comparison of GSI ',
+                                                  'statistics for one or more ',
+                                                  'experiments compared to a ',
+                                                  'reference control run.'))
+    parser.add_argument('-g','--gsistats',help='list of directories containing GSI stat files',nargs='+',required=True)
```
Comment (aerorahul) [inline scripts/plot_gsi_stat_exp.py:99]: does this need to be `-g`? It is directories containing GSI stat files; `-d` not good enough?

File: scripts/plot_gsi_stat_exp.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
```
Comment (aerorahul) [inline scripts/plot_gsi_stat_exp.py:1]: Pass this through pep8.

**Review by aerorahul — COMMENTED** [review]
looks good, but I did not test.
some minor comments.

File: scripts/plot_gsi_stat_exp.py (hunk 3 above)
Comment (CoryMartin-NOAA) [reply scripts/plot_gsi_stat_exp.py:1]: Done.

File: scripts/plot_gsi_stat_exp.py (hunk 2 above)
Comment (CoryMartin-NOAA) [reply scripts/plot_gsi_stat_exp.py]: Yeah -d works for me, fixed.

File: scripts/plot_gsi_stat_exp.py (hunk 1 above)
Comment (CoryMartin-NOAA) [reply scripts/plot_gsi_stat_exp.py]: Ok added this.

**Review by aerorahul — APPROVED** [review]
looks good, but I didn't test this.

**Comment by CoryMartin-NOAA** [comment]
I tested and put an output figure in issue #30 but I would like @CatherineThomas-NOAA to test/review before we merge in.

**Review by CatherineThomas-NOAA — APPROVED** [review]
Changes look good to me.  I tested it on Hera, compared to a plot from original script, and it works as expected.  See the attached pdf for a comparison.

[Gsistat utility.pdf](https://github.com/NOAA-EMC/PyGSI/files/5850860/Gsistat.utility.pdf)

**Comment by CoryMartin-NOAA** [comment]
Thanks @CatherineThomas-NOAA ! I'll merge this in.

**Pull request merged**